test/test_contrib.py

```diff
@@ -36,6 +36,90 @@ def test_normalize(self, device, dtype):
         assert_close(out.max(), expected_max)
 
 
+class TestKMeans(BaseTester):
+    @pytest.mark.parametrize("num_clusters", [3, 10, 1])
+    @pytest.mark.parametrize("tolerance", [10e-4, 10e-5, 10e-1])
+    @pytest.mark.parametrize("max_iterations", [10, 1000])
+    def test_smoke(self, device, dtype, num_clusters, tolerance, max_iterations):
+        N = 1000
+        D = 2
+
+        kmeans = kornia.contrib.KMeans(num_clusters, None, tolerance, max_iterations, device, 0)
+        kmeans.fit(torch.rand((N, D), dtype=dtype))
+
+        out1 = kmeans.get_cluster_assignments()
+        out2 = kmeans.get_cluster_centers()
+
+        # output is of type tensor
+        assert isinstance(out1, torch.Tensor)
+        assert isinstance(out2, torch.Tensor)
+
+        # output is of same dtype
+        assert out1.dtype == torch.int64
+        assert out2.dtype == dtype
+
+    @pytest.mark.parametrize("num_clusters", [3, 10, 1])
+    @pytest.mark.parametrize("tolerance", [10e-4, 10e-5, 10e-1])
+    @pytest.mark.parametrize("max_iterations", [10, 1000])
+    def test_cardinality(self, device, dtype, num_clusters, tolerance, max_iterations):
+        N = 1000
+        D = 2
+
+        kmeans = kornia.contrib.KMeans(num_clusters, None, tolerance, max_iterations, device, 0)
+        kmeans.fit(torch.rand((N, D), dtype=dtype))
+
+        out1 = kmeans.get_cluster_assignments()
+        out2 = kmeans.get_cluster_centers()
+
+        # output is of correct shape
+        assert out1.shape == (N,)
+        assert out2.shape == (num_clusters, D)
+
+    def test_exception(self, device, dtype):
+        # case: cluster_center = 0:
+        with pytest.raises(Exception) as errinfo:
+            kornia.contrib.KMeans(0, None, 10e-4, 10, device, 0)
+        assert "num_clusters can't be 0" in str(errinfo)
+
+        # case: cluster centers is not a 2D tensor
+        with pytest.raises(TypeError) as errinfo:
+            starting_centers = torch.rand((2, 3, 5), device=device, dtype=dtype)
+            kmeans = kornia.contrib.KMeans(None, starting_centers, 10e-4, 100, device, 0)
+        assert "shape must be [[\'C\', \'D\']]. Got torch.Size([2, 3, 5])" in str(errinfo)
+
+        # case: input data is not a 2D tensor
```

Comment:
we should adopt this, add comments for these exception tests, the tests will be more friendly for new contributors